core/src/main/resources/hudson/PluginManager/_updateSite.js

```diff
@@ -0,0 +1,6 @@
+document.getElementById("reset-to-default").onclick = function (event) {
+  event.preventDefault();
+  const siteUrlInput = document.getElementById("update-site-url");
+  siteUrlInput.value = "https://updates.jenkins.io/current/update-center.json";
```

Comment:
This may or may not be a default. so assuming it is the default would be wrong.

There may be one or more sites configured, this is called out in the Javadoc and installed system properties and/or plugins can (and do) remove / change the default UpdateSite as an when they are started.
This would break the places where it happens.

if this is to be included it should only be performed if there is a [single site](https://javadoc.jenkins.io/hudson/model/UpdateCenter.html#getSites()) defined and its [id](https://javadoc.jenkins.io/hudson/model/UpdateSite.html#getId()) is the generic [predefined one](https://javadoc.jenkins.io/hudson/model/UpdateCenter.html#PREDEFINED_UPDATE_SITE_ID).

Reply:
The predefined value will be overwritten when a user sets a new URL. It seems that a new API needs be developed. (But not sure if it's possible to get the default one after it is overwritten)

Reply:
or you could only display the `reset to default` if the above conditions are met (no new APIs are needed to see if this is the case, but you may need something to expose to the jelly to conditionally show it)

Reply:
@jtnord  I have made some changes. Now the button will be shown if there's only one site configured.